common.gypi

```diff
@@ -15,6 +15,7 @@
       }
     }]
   ],
+  'defines': [ 'NODE_API_SWALLOW_UNTHROWABLE_EXCEPTIONS' ],
```

Comment:
This should not be here by default right ?

Reply:
Hi @mhdawson ,

So this relates to my comment on https://github.com/nodejs/node-addon-api/pull/902#issuecomment-817291009 where:
> Not having the define in binding.gyp and using #define prior to #include <napi.h>: I was not able to opt-in.

You can see in commit https://github.com/nodejs/node-addon-api/pull/975/commits/a089697d9e1e37d83b834a110c585fed06d8538b the change to move the setting from `binding.gyp` to `#define` and now _all_ of the builds fail.

I do not think it is possible to have a per-compilation unit setting for this feature, because the `Error::ThrowAsJavaScriptException` is a compiled, exported symbol:

```
➜  test git:(swallow-unthrowable-exceptions) ✗ nm -C build/Release/binding_noexcept.node | grep ThrowAsJavaScriptException
00000000000024f0 t Napi::Error::ThrowAsJavaScriptException() const
```

So, when the first compilation unit uses it, it will be compiled with whatever preprocessor definitions are currently in scope.